exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFileSystem.java

```diff
@@ -83,28 +87,60 @@
   private final OperatorStats operatorStats;
   private final CompressionCodecFactory codecFactory;
 
+  private boolean initialized = false;
+
   public DrillFileSystem(Configuration fsConf) throws IOException {
     this(fsConf, null);
   }
 
   public DrillFileSystem(Configuration fsConf, OperatorStats operatorStats) throws IOException {
+    Preconditions.checkNotNull(fsConf);
+
+    // Configuration objects are mutable, and the underlying FileSystem object may directly use a passed in Configuration.
+    // In order to avoid scenarios where a Configuration can change after a DrillFileSystem is created, we make a copy
+    // of the Configuration.
+    fsConf = new Configuration(fsConf);
     this.underlyingFs = FileSystem.get(fsConf);
+
+    logger.trace("Configuration for the DrillFileSystem {} {}, underlyingFs: {}",
+      fsConf.getTrimmed(FS_DEFAULT_NAME),
+      fsConf.getTrimmed(FS_DEFAULT_NAME_KEY),
+      this.underlyingFs.getUri());
+
+    if (logger.isTraceEnabled()) {
+      logger.trace("Who made me? {}", new RuntimeException("Who made me?"));
```

Comment:
I assume it was a temp change?

Reply:
Thanks for catching this. I'll remove it after I do an RCA.